docs/source/core_concepts/index.rst

```diff
@@ -0,0 +1,522 @@
+Core concepts in k2
+===================
+
+Please refer to :cite:`mohri1997finite`, :cite:`mohri2002weighted`, and
+:cite:`mohri2008speech` for an introduction about weighted finite state
+acceptor (WFSA) and weighted finite state transducer (WFST).
+
+We use FSA to indicate either WFSA or WFST in k2.
+
+
+A simple FSA example
+---------------------
+
+A simple FSA is shown in :numref:`a simple fsa`.
+
+.. _a simple fsa:
+.. figure:: images/simple_fsa.svg
+    :alt: A simple FSA in k2
+    :align: center
+    :figwidth: 600px
+
+    A simple FSA in k2.
+
+- It has three states: 0, 1, and 2. State 0 is the start state
+  and state 2 is the final state.
+- There are 2 arcs. The first arc is from state 0 to state 1
+  with label 10 and score 0.1. The second arc is from state 1 to
+  the final state 2 with label -1 and score 0.2.
+
+.. HINT::
+
+  We use arc weight and arc score interchangeably in k2.
+
+The above FSA is created with the following code:
+
+.. code-block:: python
+
+    import k2
+    s = '''
+    0 1 10 0.1
+    1 2 -1 0.2
+    2
+    '''
+    fsa = k2.Fsa.from_str(s)
+    fsa.draw('simple_fsa.svg')
+
+We summarize the **unique** features of FSA in k2 in below:
+
+  - There is only one start state
+  - The start state is **always** 0
+  - All other states have a state number greater than 0
+  - There is only one final state
+  - The final state **always** has the **largest** state number
+  - Arcs entering the final state **always** have -1 as the label
+  - Arcs that do not enter the final state cannot have -1 as the label
+  - States have no scores
+  - All scores are on the arcs
+  - We store weights in the **positive** sense rather than as costs
+
+    .. CAUTION::
+
+        We store them as log-probs rather than negative log-probs,
+        and call them ``"scores"`` to indicate this.
+
+        They can come directly from the output of a log-softmax layer.
+
+.. HINT::
+
+  Different from other frameworks, FSAs in k2 have only a single final state.
+
+  If you want to convert an FSA from another framework to k2 that contains
```

Comment:
Just a suggestion, it's better if we can add the link of the code that how we convert opneFST format to k2 format

Reply:
Added.